### PR TITLE
feat: filter workflows based on workflow status state

### PR DESCRIPTION
**Pull request by mathiaHT**

### CODE REVIEW

File: lib/step_flow/workflows/workflows.ex [hunk 1]
```diff
@@ -191,8 +147,8 @@ defmodule StepFlow.Workflows do
         query
 
       date_value ->
-        date = Date.from_iso8601!(date_value)
-        from(workflow in query, where: fragment("?::date", workflow.inserted_at) <= ^date)
+        date = NaiveDateTime.from_iso8601!(date_value)
```
Comment (MarcAntoine-Arnaud) [inline lib/step_flow/workflows/workflows.ex:150]: to keep retro-compatibility, you may need to do:
```elixir
date =
  case NaiveDateTime.from_iso8601!(date_value) do
    {:ok, date} -> date
   _ -> Date.from_iso8601!(date_value)
  end
```

File: lib/step_flow/workflows/workflows.ex [hunk 2]
```diff
@@ -202,8 +158,8 @@ defmodule StepFlow.Workflows do
         query
 
       date_value ->
-        date = Date.from_iso8601!(date_value)
-        from(workflow in query, where: fragment("?::date", workflow.inserted_at) >= ^date)
+        date = NaiveDateTime.from_iso8601!(date_value)
```
Comment (MarcAntoine-Arnaud) [inline lib/step_flow/workflows/workflows.ex:161]: same here

**Review by MarcAntoine-Arnaud — CHANGES_REQUESTED** [review]

File: lib/step_flow/workflows/workflows.ex (hunk 1 above)
Comment (mathiaHT) [reply lib/step_flow/workflows/workflows.ex:150]: what if I check all usage of the endpoint using the function in the backend?

File: lib/step_flow/workflows/workflows.ex [hunk 3]
```diff
@@ -191,8 +146,13 @@ defmodule StepFlow.Workflows do
...
       date_value ->
-        date = Date.from_iso8601!(date_value)
-        from(workflow in query, where: fragment("?::date", workflow.inserted_at) <= ^date)
+        date =
+          case NaiveDateTime.from_iso8601(date_value) do
+            {:ok, date} -> date
+            _ -> Date.from_iso8601!(date_value)
+          end
+
+        from(workflow in query, where: fragment("?::timestamp", workflow.inserted_at) <= ^date)
```
Comment (MarcAntoine-Arnaud) [inline lib/step_flow/workflows/workflows.ex:155]: as you've changed from `date` to `timestamp`, the `Date.from_iso8601` is incorrect. You need to convert into a `DateTime` before

File: lib/step_flow/workflows/workflows.ex [hunk 4]
```diff
@@ -202,8 +162,13 @@ defmodule StepFlow.Workflows do
         query
 
       date_value ->
-        date = Date.from_iso8601!(date_value)
-        from(workflow in query, where: fragment("?::date", workflow.inserted_at) >= ^date)
+        date =
+          case NaiveDateTime.from_iso8601(date_value) do
+            {:ok, date} -> date
+            _ -> Date.from_iso8601!(date_value)
```
Comment (MarcAntoine-Arnaud) [inline lib/step_flow/workflows/workflows.ex:168]: same here

**Review by MarcAntoine-Arnaud — CHANGES_REQUESTED** [review]

File: lib/step_flow/workflows/workflows.ex [hunk 5]
```diff
@@ -162,13 +168,19 @@ defmodule StepFlow.Workflows do
...
           case NaiveDateTime.from_iso8601(date_value) do
-            {:ok, date} -> date
-            _ -> Date.from_iso8601!(date_value)
+            {:ok, date} ->
+              date
+
+            _ ->
+              NaiveDateTime.new!(
+                Date.from_iso8601!(date_value),
+                Time.new!(23, 59, 59, 999_999)
```
Comment (MarcAntoine-Arnaud) [inline lib/step_flow/workflows/workflows.ex:179]: for me the old version will include the day, so it requires to update with:
```elixir
NaiveDateTime.new!(
  Date.from_iso8601!(date_value),
  Time.new!(0, 0, 0, 0)
)
```

**Review by MarcAntoine-Arnaud — CHANGES_REQUESTED** [review]

**Pull request merged**